File: Tools/GDSTranslator/tools/tmp/output/csharp/parameter.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace GDSKit
{
	public sealed partial class parameter
	{
		private static Dictionary<string, parameter> _dict_ = new Dictionary<string, parameter>();

		public string parameter_name;
		public string parameter_value;
		

		private static bool _isInited_ = false;

		public static void Initialize(List<string[]> data)
		{
			if (_isInited_) return;

			foreach (var objArr in data)
			{
				parameter _value_ = CreateInstance(objArr);
				PostProcessor(_value_);
				string _key_ = _value_.GetKeyInDict();

				_dict_.Add(_key_, _value_);
			}

			_isInited_ = true;
		}

		private static parameter CreateInstance(string[] objArr)
		{
			parameter _ret_ = new parameter();
			
			_ret_.parameter_name = String.Intern(CSVParser.GetAsString(objArr[0]));
			
			_ret_.parameter_value = String.Intern(CSVParser.GetAsString(objArr[1]));
			
			

			return _ret_;
		}

		static partial void PostProcessor(parameter instance);

		private string GetKeyInDict()
		{
			return this.parameter_name.ToString();
		}

		public delegate bool parameterFilter(parameter obj);
		public static List<parameter> Find(parameterFilter filter)
		{
			if (null != _externalDataSource_)
			{
				throw new NotImplementedException();
			}

			List<parameter> _ret_ = new List<parameter>();

			foreach(KeyValuePair<string, parameter> kvp in _dict_)
			{
				if (filter.Invoke(kvp.Value))
				{
					_ret_.Add(kvp.Value);
				}
			}

			return _ret_;
		}
		
		public static bool HasInstance(string parameter_name)
		{
			string _key_ = parameter_name.ToString();
			return _dict_.ContainsKey(_key_);
		}

		public static void AddInstance(parameter obj)
		{
			string _key_ = obj.GetKeyInDict();
			if (_dict_.ContainsKey(_key_))
			{
				UnityEngine.Debug.LogWarning("在名为parameter的GDS文件中已经存在: " + _key_);
				return;
			}
			_dict_.Add(_key_, obj);
		}

		public static parameter GetInstance(string parameter_name)
		{
			if (!_isInited_)
			{
				return null;
			}
			string _key_ = parameter_name.ToString();
			if (null != _externalDataSource_)
			{
				return _externalDataSource_.Invoke(typeof(parameter), _key_) as parameter;
			}
			else
			{
				parameter _ret_ = null;
				if (!_dict_.TryGetValue(_key_, out _ret_))
				{
					throw new Exception("在名为parameter的GDS文件中没有发现key : " + _key_);
				}
				return _ret_;
			}
		}

		private static Func<System.Type, string, object> _externalDataSource_;
		public static void TransitionToExternalMode(Func<System.Type, string, object> externalHook, 
													bool doDump, 
													Action<System.Type, List<string>, List<object>> externalDumper)
		{
			_externalDataSource_ = externalHook;
			if (doDump)
			{
				List<string> keys = new List<string>();
				List<object> values = new List<object>();
				foreach (var kvp in _dict_)
				{
					keys.Add(kvp.Key);
					values.Add(kvp.Value);
				}
				externalDumper.Invoke(typeof(parameter), keys, values);
			}
			_dict_.Clear();
		}

		public static int GetCount()
		{
			return _dict_.Count;
		}
	}
}

File: Tools/GDSTranslator/tools/tmp/output/golang/parameter/parameter.go
package parameter

import (
	"fmt"
	"strconv"
	"strings"
)



type Parameter struct {
    ParameterName   string
    ParameterValue  string

}

type ParameterDict map[string]Parameter

var (
	dict     map[string]ParameterDict
	isInited bool = false
)

func Initialize(ver string, data [][]interface{}) {

	i, _ := strconv.Atoi("12345")

	i += 1
	
	if !isInited {
		dict = make(map[string]ParameterDict)	
		isInited = true
	}

	__dict := make(map[string]Parameter)
	for _, _strArray_ := range data {
		_instance_ := CreateInstance(_strArray_)
		key := _instance_.getKeyInDict()
		__dict[key] = _instance_
	}

	dict[ver] = __dict
}

func CreateInstance(_data_ []interface{}) Parameter {
	_ret_ := Parameter{}

    _ret_.ParameterName = strings.TrimSpace(_data_[0].(string))

    _ret_.ParameterValue = strings.TrimSpace(_data_[1].(string))

	

	return _ret_
}

func GetInstance(ver string, ParameterName string) *Parameter {
	if !isInited {
		return nil
	}
	__dict := dict[ver]
	_key_ := fmt.Sprint(ParameterName)
	_ret_ := __dict[_key_]
	return &_ret_
}

func (_self_ *Parameter) getKeyInDict() string {
	return fmt.Sprint(_self_.ParameterName)
}

type ParameterFilter func(b Parameter) bool

func Find(ver string, filter ParameterFilter) []Parameter {
	ret := make([]Parameter, 0)

	__dict := dict[ver]
	for _, val := range __dict {
		if filter(val) {
			ret = append(ret, val)
		}
	}

	return ret
}
